### PR TITLE
Teachers being able to assign students to peer review each other (making groups)

### LINKED ISSUE #67: Task.22 - Teachers being able to assign students to peer review each other (making teams)
Teachers being able to assign students to peer review each other (making teams)

The teacher should be able to specify a number of student per group. Then students will be grouped randomly according to that number of max members of the group.

### PR DESCRIPTION
**Pull request by sasha-klnd**
Closes branch Task.22, allowing teachers to create groups and assign students

Currently the creating groups functionality is implemented, and the adding students functionality is being worked on.

### CODE REVIEW

**Comment by Marian-Hristov** [comment]
@sasha-klnd update the description of your PR, to be more descript in addition to its current state

File: client/src/components/teacher/TeacherView.jsx [hunk 1]
```diff
@@ -43,6 +46,10 @@ function TeacherView() {
     setCourses([...courses, newCourse]);
   };
 
+  const handleGroupAddition = (newGroup) => {
+    setGroups([...groups, newGroup]);
```
Comment (Marian-Hristov) [inline client/src/components/teacher/TeacherView.jsx:66]: good with updating the UI without having to refetch on the parent component

File: client/src/components/teacher/TeacherView.jsx [hunk 2]
```diff
@@ -56,13 +63,47 @@ function TeacherView() {
...
+        break;
+      } 
+      case 'group': {
+        setIsCreatingGroup(false);
+        break;
+      }
+      default: break;
+    }
+    return;
+  }
```
Comment (Marian-Hristov) [inline client/src/components/teacher/TeacherView.jsx:125]: good handling of different modals open/close function while keeping a single function

File: client/src/components/teacher/TeacherView.jsx [hunk 3]
```diff
@@ -74,8 +115,23 @@ function TeacherView() {
             {selectedCourse.dept} {selectedCourse.number}
           </h1>
           <div style={{ display: "flex", gap: "10px" }}>
-            <button className="otherbtn">Create Teams</button>
+            <button className="otherbtn"
+              onClick={async (e) => {
+                e.preventDefault();
+                // handleOpen('group');
```
Comment (Marian-Hristov) [inline client/src/components/teacher/TeacherView.jsx]:
```suggestion
```
remove this comment please

**Review by Marian-Hristov — CHANGES_REQUESTED** [review]
very nice work! keep it going!

**Comment by sasha-klnd** [comment]
Unfortunately I have an orthodontist appointment which will delay my work, do you think I could receive some help on implementing the students being added to groups functionality @Marian-Hristov or @Vinuyans?

**Review by Vinuyans — APPROVED** [review]
lgtm

**Review by Marian-Hristov — APPROVED** [review]

**Pull request merged**